src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -857,6 +857,15 @@ public void switchFolder(FolderElement switchTo) {
 
 		upFolder.setEnabled(!currentFolder.isRoot());
 	}
+	
+	private void togglefilter(String filter) {
+		if (!Pattern.matches(".*" + filter + ".*", search.getText())) {
+			search.setText(search.getText() + " " + filter);
+		} else {
+			search.setText(search.getText().replaceAll(" " + filter, ""));

```

Comment:
Move both replaces in single line, you can chain replace statements, also use replaceAll only when you use regex. Also use regex to remove multiple spaces, not hardcoded two spaces

Reply:
like this?
search.setText(search.getText().replaceAll(" " + filter, "").replaceAll("  ", " "));